### ISSUE TITLE
assert error in pydrive2.fs.GDriveFileSystem.get_file

### ISSUE THREAD

**Issue by phpjunkie420** [iterative/PyDrive2#359]
I've look at the source for this and the lpath doesn't start with root it throws an AssertionError.

```python
            drive = GoogleDrive()  # inherits pydrive2.fs.GDriveFileSystem

            with TableProgress(transient = False) as progress:  # inherits rich.progress.Progress to overwrite def get_renderables(self) to wrap a table around the progress bar
                info = drive.info(f'root/SpyBot/10-30-2024 12.31.35.zip')
                task_id = progress.add_task(description = '', total = info['size'])

                def callback_handler(_, value: int):
                    progress.update(task_id = task_id, completed = info['size'])

                with fsspec.callbacks.Callback(
                        hooks = {
                            'progress': callback_handler
                        }
                ) as callback:
                    drive.get_file(lpath = rf'C:\Users\phpjunkie\Python\Scripts\scratches\temp{os.path.basename(info['name'])}', rpath = info['name'], block_size = 1 << 20, callback = callback)
                sleep(1)
```
```
  File "C:\Users\phpjunkie\AppData\Local\Programs\Python\Python312\Lib\site-packages\pydrive2\fs\spec.py", line 372, in _get_item_id
    assert bucket == self.root
           ^^^^^^^^^^^^^^^^^^^
AssertionError
```
If I change `lpath = rf'C:\Users\phpjunkie\Python\Scripts\scratches\temp{os.path.basename(info['name'])}',` to `lpath = info['name'],` with . . .
```python
os.makedirs(os.path.dirname(info['name']), exist_ok = True)
if os.path.exists(info['name']):
    os.remove(info['name'])
```
before it to create the folder path, everything works just file. The `assert bucket == self.root` is obviously checking if both the `lpath`, and `rpath` start with root in the folder path.

**Comment by shcheklein** [comment]
@phpjunkie420 I've published a new version 1.21.1 - please give it a try. I think the reason was in wrong naming of the args. I don't see tbh that it is doing a check (assert) on the `lpath` now. But let me know, and we can fix it separately (the PR I made still makes sense).

**Comment by phpjunkie420** [comment]
Tested and thank you for the fix.
